module.go

```diff
@@ -21,9 +28,94 @@ type Scanner interface {
 	Protocol() string
 
 	// Scan connects to a host. The result should be JSON-serializable
-	Scan(t ScanTarget) (ScanStatus, any, error)
+	Scan(ctx context.Context, t *ScanTarget, dialer *DialerGroup) (ScanStatus, any, error)
```

Comment:
This is beyond nitpicky, but would move general stuff before the ScanTarget unless the DialerGroup is optional